### PR TITLE
WT-10346 Create unit testing framework for Azure

### PR DESCRIPTION
**Pull request by mariammojid**

### CODE REVIEW

File: ext/storage_sources/azure_store/test/test_azure_connection.cpp [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#define CATCH_CONFIG_MAIN
+#include <catch2/catch.hpp>
+
+#include "azure_connection.h"
+
+TEST_CASE("testing class azure_connection", "azure_connection")
+{}
```
Comment (jiechenbo) [inline ext/storage_sources/azure_store/test/test_azure_connection.cpp:35]: New line at the end of this file, and i would place the curly braces on the same line.

**Review by jiechenbo — CHANGES_REQUESTED** [review]
Close to LGTM

File: ext/storage_sources/azure_store/test/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    run_azure_unit_tests
+    PRIVATE ${COMPILER_DIAGNOSTIC_CXX_FLAGS}
+)
+
+target_link_libraries(run_azure_unit_tests Catch2::Catch2)
+
+# Register the test with CTest.
+add_test(NAME azure_unit_tests COMMAND run_azure_unit_tests)
+# Label the tests so they can be run explicitly.
+set_tests_properties(azure_unit_tests PROPERTIES LABELS "azure_store")
```
[the remark this answers is not in the document]
Comment (mariammojid) [reply ext/storage_sources/azure_store/test/CMakeLists.txt]: done :)

File: ext/storage_sources/azure_store/test/test_azure_connection.cpp (hunk 1 above)
Comment (mariammojid) [reply ext/storage_sources/azure_store/test/test_azure_connection.cpp]: fixed 👍🏻

**Review by jiechenbo — APPROVED** [review]
LGTM!

**Review by korteland — COMMENTED** [review]
This is OK as-is, but did you consider adding these tests to the existing unit tests at any point? It would save a lot of extra CMake overhead since we're not linking against them twice, and the labelling thing would be a little less weird (there are two levels of labelling right now, one for `azure_store` at the ctest level and one for `azure_connection` at the catch2 level).

**Comment by jiechenbo** [comment]
@korteland hmm thank for you the question, i think you made a really good point here. I have followed the design structure of S3, because they also have unit tests for their implementation inside the directory.  There were two reasons for this, the unit test aims to only test the connection class portion of the test. Is it possible to include only that connection class file in the unittest directory? And for the second reason it seems that the unit test directories directly test WT functions, where as this is planning to test outside of WT. What are your thoughts?

**Comment by korteland** [comment]
> Is it possible to include only that connection class file in the unittest directory?

I'm not sure what you mean here sorry, are you talking about `azure_connection.h` being included in the unittest directory? Because I don't think that would change, you'd still have to include it wherever you're testing it.

> it seems that the unit test directories directly test WT functions, where as this is planning to test outside of WT

Yeah that's a good point. Given that this is an extension, it seems cleaner to keep any changes outside the main codebase. I think this design is the best way to do that, I'll do a more detailed re-review given that this overall approach is good.

**Review by korteland — APPROVED** [review]
LGTM!

**Pull request merged**